src/main/java/org/traccar/helper/IpRetriever.java

```diff
@@ -0,0 +1,50 @@
+package org.traccar.helper;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Gets the client's IP address regardless of whether
+ * the server is behind a proxy/reverse proxy server or a load balancer.
+ */
+public final class IpRetriever {
+
+    /**
+     * Retrieves the client's IP address.
+     * Handles the cases like whether the server is behind a proxy/reverse proxy server or a load balancer
+     *
+     * @param request {@link HttpServletRequest} instance
+     * @return client's IP address
+     */
+    public static String retrieveIP(HttpServletRequest request) {
```

Comment:
I would can this `retrieveRemoteAddress` or something like that.

Reply:
Hmm, yeah, sounds better.